rust/azure_iot_operations_mqtt/src/session/wrapper.rs

```diff
@@ -72,10 +80,15 @@ impl Session {
         let client_id = options.connection_settings.client_id.clone();
         let sat_file = options.connection_settings.sat_file.clone();
 
-        // Add AIO metric to user properties when using AIO MQTT broker features
+        // Add AIO metric and features to user properties when using AIO MQTT broker features
         // TODO: consider user properties from being supported on SessionOptions or ConnectionSettings
-        let user_properties = if options.aio_broker_features {
-            vec![("metriccategory".into(), "aiosdk-rust".into())]
+        let user_properties = if let Some(features) = options.aio_broker_features {
+            let mut user_properties =
+                vec![("metriccategory".to_string(), "aiosdk-rust".to_string())];
+            if features.persistence {
+                user_properties.push(("aio-persistence".to_string(), true.to_string()));
```

Comment:
Is my understanding correct that this enables the feature for the conection, but the more granular usage from the `services` implementation below does it per-message?